airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/config/AirbyteConfiguredCatalog.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.config
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import io.airbyte.commons.json.Jsons
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog
+import io.micronaut.context.annotation.ConfigurationProperties
+import io.micronaut.core.util.StringUtils
+
+/**
+ * Micronaut configured properties holder for the Airbyte configured catalog provided to the
+ * connector CLI as an argument.
+ */
+@ConfigurationProperties("airbyte.connector.catalog")
+@SuppressFBWarnings(
+    value = ["NP_NONNULL_RETURN_VIOLATION"],
+    justification = "Uses dependency injection",
+)
+class AirbyteConfiguredCatalog {
+    lateinit var configured: String
+
+    fun getConfiguredCatalog(): ConfiguredAirbyteCatalog {
```

Comment:
Consider just calling this `get()`? I may be wrong but at the call site it should be fairly obvious that this object is a no-frills wrapper object.

Reply:
@postamar I'll give this a try, but I'm not sure that this will work with the Micronaut configuration binding stuff.  I think that I had to make it a named method to avoid some issues there.

Reply:
I remember why this doesn't work.  The data coming from Micronaut needs to be a String for it to be able to inject it, but we want to return a `ConfiguredAirbyteCatalog` for the code, hence the separate getter method from the field.

Reply:
Thanks for looking into it!